pgctl/cli.py

```diff
@@ -238,6 +238,11 @@ def __change_state(self, state):
                     for service in self.all_services
                 )
 
+            # Run the playground-wide "post-start" hook (if it exists) upon every `pgctl start'
+            if state is Start:
+                if all(service.state['state'] == 'ready' for service in self.services):
+                    self._run_playground_wide_hook('post-start')
```

Comment:
What's the reason for running this while holding the lock? Originally we were concerned that holding the lock would lead to a confusing UX for developers if for some reason the hook script hangs for a long time. I'm wondering if we should apply the same logic here, or if there's a reason it makes sense to hold the lock in this case?